### ISSUE TITLE
Always "Nothing"

### ISSUE THREAD

**Issue by a626677909** [asingh33/CNNGestureRecognizer#39]
![手势识别](https://user-images.githubusercontent.com/19469719/54086604-c406c480-4385-11e9-9673-7b6ce67791eb.png)

**Comment by asingh33** [comment]
Hey,
There are too many things in your background as per the image. That would affect the image being fed into the CNN. Binary mode wont help much.
I would suggest:
- To try with a relatively lesser things in the background say a simple wall.
- Try with Skin mode by pressing 'b' by highlighting the 'Original' window. If lighting is good then it should work well.
- You can even try once with Background submask mode by pressing 'x' key. For this first press x by not putting your hand. And once you see the ROI window completely blank then its ready for use.

try and tell me

**Comment by imrahul13** [comment]
Hey, first of all congrats for your awesome work. But i am also facing the same always nothing problem even after following above 3 points... plz help

**Comment by Ivancc99** [comment]
I have the same problem, and I fixed it.
The solution is to train a new model（run the program and choose the second option）.

**Comment by a626677909** [comment]
> Hey,
> There are too many things in your background as per the image. That would affect the image being fed into the CNN. Binary mode wont help much.
> I would suggest:
> 
> * To try with a relatively lesser things in the background say a simple wall.
> * Try with Skin mode by pressing 'b' by highlighting the 'Original' window. If lighting is good then it should work well.
> * You can even try once with Background submask mode by pressing 'x' key. For this first press x by not putting your hand. And once you see the ROI window completely blank then its ready for use.
> 
> try and tell me

Well.... none of these methods worked.  But after I retrained the model, it worked pretty well. I think there must be something worng with the .hdf5 document on you Google Drive.  And Thanks for your helping.

**Comment by imrahul13** [comment]
> I have the same problem, and I fixed it.
> The solution is to train a new model（run the program and choose the second option）.

I got the following error while i chose to train new model(option 2)..
WARNING:tensorflow:From C:\Users\pavilion\Anaconda3\lib\site-packages\tensorflow\python\framework\op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.

Please help.

**Comment by jhonatantirado** [comment]
> > I have the same problem, and I fixed it.
> > The solution is to train a new model（run the program and choose the second option）.
> 
> I got the following error while i chose to train new model(option 2)..
> WARNING:tensorflow:From C:\Users\pavilion\Anaconda3\lib\site-packages\tensorflow\python\framework\op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
> Instructions for updating:
> Colocations handled automatically by placer.
> 
> Please help.

That's not an error. It's just a warning message. I was able to continue. Just press ENTER to continue the training process.

**Comment by jhonatantirado** [comment]
It works properly after retraining the model. It took about 2 hours on my Intel Core i7 / 8GB RAM laptop. I used the Background filter and the SkinMask mode.

![image](https://user-images.githubusercontent.com/876293/58359605-e74ff480-7e49-11e9-9d89-a981b7140141.png)

**Comment by asingh33** [comment]
Closing this issue as retraining helps.
I have recently found a new observation that same pretrained weight file doesn't work across the platform. The default weight files I had earlier provided with the repo was on Windows. So it won't work as it is for macOS or Linux.
Today I have provided links to two macOS and Windows specific weight files.
Pull the latest code and try these weight file or retrain. It will work.
